File: src/main/java/com/freya02/botcommands/api/pagination/interactive/SelectContent.java
package com.freya02.botcommands.api.pagination.interactive;


import net.dv8tion.jda.api.entities.Emoji;
import net.dv8tion.jda.api.interactions.components.selections.SelectOption;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public record SelectContent(@NotNull String label, @Nullable String description, @Nullable Emoji emoji) {
	@NotNull
	SelectOption toSelectOption(@NotNull String value) {
		return SelectOption.of(label, value).withDescription(description).withEmoji(emoji);
	}

	@NotNull
	public static SelectContent of(@NotNull String label) {
		return new SelectContent(label, null, null);
	}

	@NotNull
	public static SelectContent of(@NotNull String label, @Nullable String description) {
		return new SelectContent(label, description, null);
	}

	@NotNull
	public static SelectContent of(@NotNull String label, @Nullable String description, @Nullable Emoji emoji) {
		return new SelectContent(label, description, emoji);
	}
}

File: src/main/kotlin/io/github/freya022/botcommands/api/pagination/interactive/SelectContent.kt
package io.github.freya022.botcommands.api.pagination.interactive

import net.dv8tion.jda.api.entities.emoji.Emoji
import net.dv8tion.jda.api.interactions.components.selections.SelectOption

data class SelectContent(val label: String, val description: String?, val emoji: Emoji?) {
    fun toSelectOption(value: String): SelectOption {
        var selectOption = SelectOption.of(label, value)
        description?.let { selectOption = selectOption.withDescription(description) }
        emoji?.let { selectOption = selectOption.withEmoji(emoji) }

        return selectOption
    }

    companion object {
        @JvmStatic
        @JvmOverloads
        fun of(label: String, description: String? = null, emoji: Emoji? = null): SelectContent {
            return SelectContent(label, description, emoji)
        }
    }
}
